drivers/watchdog/wdt_wwdg_stm32.c

```diff
@@ -114,10 +114,17 @@ static uint32_t wwdg_stm32_get_timeout(const struct device *dev,
 				       uint32_t prescaler_exp,
 				       uint32_t counter)
 {
-	uint32_t divider = WWDG_INTERNAL_DIVIDER * (1 << prescaler_exp);
-	float f_wwdg = (float)wwdg_stm32_get_pclk(dev) / divider;
+	uint32_t pclk = wwdg_stm32_get_pclk(dev);
+	uint32_t ticks = ((counter & 0x3FU) + 1U);
+	uint64_t divider = WWDG_INTERNAL_DIVIDER * (1U << prescaler_exp);
 
-	return USEC_PER_SEC * (((counter & 0x3F) + 1) / f_wwdg);
+	/*
+	 * timeout formula reordered to increase precision without floats
+	 * time_us = floor(ticks * divider * USEC_PER_SEC / pclk).
+	 */
```

Comment:
Inline comment is not needed IMHO. The commit message already describes the change.